### PR TITLE
Tests and smart contract refactor

### PR DESCRIPTION
**Pull request by nlukic97**
- implemented waffle fixtures
- modular refactor of tests into unit and integration tests
- small update of smart contract logic

### CODE REVIEW

**Comment by nlukic97** [comment]
Converted to draft until we audit the contract

**Review by stevyhacker — APPROVED** [review]
This is an excellent improvement on the tests!
Admin changes look very good too 👍

File: contracts/AirbroFactory.sol [hunk 1]
```diff
@@ -116,10 +122,17 @@ contract AirbroFactory {
...
-            merkleRootHash
+            admin
         );
         airdrops.push(address(tokenDropContract));
         totalAirdropsCount++;
         emit NewAirdrop(rewardedNftCollection, address(tokenDropContract), msg.sender);
     }
+
+    /// @notice Updates the address of the admin variable
+    /// @param _newAdmin - New address for the admin of this contract, and the address for all newly created airdrop contracts
```
Comment (stevyhacker) [inline contracts/AirbroFactory.sol:133]: Airdrop contracts are not using the admin address from the factory but you pass the same address from the factory in their constructor. That is a fine implementation and I think it is sufficient for this use case but also have in mind what you suggested about referencing from the airdrop contract and pulling the admin address from the factory so the Factory contract can in one transaction replace all admin addresses for all existing airdrop contracts.

Comment (nlukic97) [reply contracts/AirbroFactory.sol:133]: That's a great suggestion, definitively should be implemented

**Pull request merged**